pkg/pgclient/client.go

```diff
@@ -114,6 +114,50 @@ func (cfg *Config) GetConnectionStr() string {
 		cfg.host, cfg.port, cfg.user, cfg.database, cfg.password, cfg.sslMode)
 }
 
+func (cfg *Config) GetNumConnections() (min int, max int, err error) {
```

Comment:
why is this modifying the cfg.ConnectionsPerProc variable? This shouldn't be called more than once correct? So this isn't about memoization?

There is some weird behavior if this is called more than once: you'll never hit a bunch of branches like line 147 on subsequent runs.

I'd rather not have this modify cfg state if possible